### ISSUE TITLE
Split assembly into files and compile during build script

### ISSUE THREAD

**Issue by nhynes** [oasisprotocol/deoxysii-rust#24]
so that everyone else can finally use rust stable

**Comment by nhynes** [comment]
[_mm256_zeroall](https://doc.rust-lang.org/core/arch/x86_64/fn._mm256_zeroall.html) was added with the AVX instruction set, which does the trick. Everyone has AVX now, right?

**Comment by Yawning** [comment]
> Everyone has AVX now, right?

No.  For example the Celeron G5900T (Comet Lake, Q2 2020) does not.

**Comment by gw0** [comment]
"Second-generation Intel® Core™ processors (Intel® microarchitecture code name Sandy Bridge), released in Q1, 2011, are the first from Intel supporting Intel® AVX technology."

**Comment by Yawning** [comment]
> "Second-generation Intel® Core™ processors (Intel® microarchitecture code name Sandy Bridge), released in Q1, 2011, are the first from Intel supporting Intel® AVX technology." ([source](https://software.intel.com/content/www/us/en/develop/articles/introduction-to-intel-advanced-vector-extensions.html))

And?  That's not useful at all, because not all CPUs support every extension. See [the Comet Lake Celerons](https://ark.intel.com/content/www/us/en/ark/products/199270/intel-celeron-processor-g5900t-2m-cache-3-20-ghz.html) (mentioned in my reply), which are 2020 product for a good example of this.  The preliminary datasheet for the J6413 (Elkhart Lake, Q1 2021) indicates that this trend will continue for that product line.

This package currently this uses SSE2, SSSE3 (`PSHUFB`), and AES-NI (`AESENC`).  While SSSE3 is also technically an optional extension, there are precisely zero CPUs that support AES-NI, that do not support SSSE3.  So to be explicitly clear, currently, "it will run on any CPU that supports AES-NI", and any changes should keep things this way.
